lib/morpheus.gi

```diff
@@ -2587,6 +2587,26 @@ InstallMethod( InnerAutomorphismsAutomorphismGroup,
     end );
 
 
+#############################################################################
+##
+#M  InnerAutomorphismGroup( <G> )
+##
+InstallMethod( InnerAutomorphismGroup,
+    "for groups",
+    true,
+    [ IsGroup and HasGeneratorsOfGroup ], 0,
+    function( G )
+    local gens;
+    if HasAutomorphismGroup( G ) or IsTrivial( G ) then
+        return InnerAutomorphismsAutomorphismGroup( AutomorphismGroup( G ) );
+    fi;
+    gens:= GeneratorsOfGroup( G );
+    # get the non-central generators
+    gens:= Filtered( gens, i -> not ForAll( gens, j -> i*j = j*i ) );
+    return Group( List( gens, i -> InnerAutomorphism( G, i ) ) );
```

Comment:
You want to call
`SetIsGroupOfAutomorphismsFiniteGroup(autgrp,true)` on this group, so GAP knows that this is a group of group automorphisms, and thus uses appropriate `Nice` methods to work with the group. (This is a pain to do, but as far as I know there is no way to deduce this information automatically from the type.)

Reply:
Thanks, done!